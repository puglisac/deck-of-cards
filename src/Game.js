import React from "react";
import Card from "./Card";
import { useState, useEffect } from "react";
import axios from "axios";

const Game = () => {
	const [ img, setImg ] = useState("");
	const [ deck, setDeck ] = useState("");
	const [ drawing, setDrawing ] = useState(false);
	const [ buttonText, setButtonText ] = useState("Draw Cards");
	useEffect(() => {
		newDeck();
	}, []);

	useEffect(
		() => {
			let id;
			if (drawing) {
				setButtonText("Pause");
				drawCard();
				id = setInterval(() => drawCard(), 1000);
			} else setButtonText("Draw Cards");
			return () => clearInterval(id);
		},
		[ drawing ]
	);

	const newDeck = async () => {
		const res = await axios.get("https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1");
		setDeck(res.data.deck_id);
	};
	const suffleDeck = async () => {
		await axios.get(`https://deckofcardsapi.com/api/deck/${deck}/shuffle/`);
		setImg("");
		setDrawing(false);
	};

	const drawCard = async () => {
		const card = await axios.get(`https://deckofcardsapi.com/api/deck/${deck}/draw/?count=1`);
		try {
			setImg(card.data.cards[0].image);
		} catch (TypeError) {
			setDrawing(false);
			alert("No more cards in deck!");
		}
	};
	return (
		<div>
			<button onClick={() => setDrawing(!drawing)}>{buttonText}</button>
			<button onClick={suffleDeck}>Suffle Deck</button>
			<div>{img ? <Card img={img} /> : null}</div>
		</div>
	);
};
export default Game;
