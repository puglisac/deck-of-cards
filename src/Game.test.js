import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Game from "./Game";

test("renders without crashing", () => {
	render(<Game />);
});
it("matches snapshot", function() {
	const { asFragment } = render(<Game />);
	expect(asFragment()).toMatchSnapshot();
});

it("can start draw cards", function() {
	const { queryByAltText, getByText } = render(<Game />);
	const drawBtn = getByText("Draw Cards");

	fireEvent.click(drawBtn);
	expect(drawBtn).toHaveTextContent("Pause");
});
