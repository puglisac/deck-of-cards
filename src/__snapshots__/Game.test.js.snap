// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Draw Cards
    </button>
    <button>
      Suffle Deck
    </button>
    <div />
  </div>
</DocumentFragment>
`;
